### PR TITLE
UI: Update ember-a11y-testing

### PR DESCRIPTION
**Pull request by backspace**
This was originally in #9641 but it became quite involved as described below, so I decided to update it in (relative) isolation.

I needed to update `ember-a11y-testing` because of failures with Ember 3.20 such as [this](https://github.com/hashicorp/nomad/runs/1728395042?check_suite_focus=true#step:4:284). The new version has a peer dependency on `@ember/test-helpers` v2 and we only had various v1s through other addons.

The `ember-fetch` update is because of [this](https://github.com/ember-cli/ember-fetch/issues/547).

It’s hard to see because `yarn.lock` thrashing is the only evidence, but I also had to `rm -rf node_modules yarn.lock` to fix a catastrophic Babel error discussed [here](https://github.com/storybookjs/storybook/issues/12893). I much prefer targeting specific dependencies for updates but I couldn’t determine which ones were the problem so I did as suggested in that thread 😞

See 9ee5c96 and 1982358 for comments on specific changes.

It seems pretty minimal but it took me surprisingly long to get to this point 😭

### CODE REVIEW

**Comment by backspace** [comment]
> ## Ember Test Audit flaky tests
> 
> Ember Test Audit detected these flaky tests on [5522229](https://github.com/hashicorp/nomad/commit/55222293ae1385efaa0f5a54e2908bc06e114ca1):
> 
>     * Acceptance | optimize search and facets: the Datacenter facet has the correct options
> 
>     * Acceptance | optimize search and facets: selecting options in the Datacenter facet updates the dc query param
> 
>     * Acceptance | optimize search and facets: selecting options in the Prefix facet updates the prefix query param

I’ve deleted the above comment because I deliberately pushed with failures, but wanted to note that the flakiness report worked 🤩

**Comment by DingoEatingFuzz** [comment]
Oof. What a journey!

File: ui/app/adapters/watchable.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default class Watchable extends ApplicationAdapter {
       signal,
       data: params,
     }).catch(error => {
-      if (error instanceof AbortError) {
+      if (error instanceof AbortError || error.name == 'AbortError') {
```
Comment (DingoEatingFuzz) [inline ui/app/adapters/watchable.js:63]: Hmm. These make me sad, but they're definitely not the end of the world. I think I'll open an issue tracking this particular thing. I wouldn't want to block the Ember and related deps work on it.

**Review by DingoEatingFuzz — APPROVED** [review]

File: ui/app/adapters/watchable.js (hunk 1 above)
Comment (backspace) [reply ui/app/adapters/watchable.js:63]: ya, I don’t love it, a new issue sounds good to me so I can proceed through this slog 😆💞

**Pull request merged**